jenkins/README.md

```diff
@@ -43,39 +37,104 @@ If the author is not in the whitelist but should be trusted then by adding a
+5. [/jenkins/scripts/integration_delete.sh](https://github.com/metal3-io/project-infra/blob/master/jenkins/scripts/integration_delete.sh)
+
+### GitHub Pull Request Builder, a.k.a. ghprb
+
+For our Jenkins to work with Github we need [GitHub Pull Request Builder](https://plugins.jenkins.io/ghprb/), a.k.a. ghprb plugin installed in Jenkins. This is already done by Nordix admins. The plugin handles everything related to your PR. In short, ghprb exposes a weebhook endpoint where GutHub webhook service can send webhook calls to. Current Jenkins ghprb webhook is exposed at https://jenkins.nordix.org/ghprbhook/, and most of the Metal3 GitHub repositories have webhooks configured to redirect webhook calls to that URL.
+When configuring GitHub webhooks: 
 
-The job configuration is stored [here](https://gerrit.nordix.org/admin/repos/infra/cicd).
-Please announce if some reviews are needed on #cluster-api-baremetal in Kubernetes
-slack or send an email to estjorvas [at] est.tech mailing list.
+1. Paste the ghprb URL in **Payload URL**;
 
-### Job pipeline
+1. select application/json for the **Content Type**;
 
-The pipeline for the jobs is in the `jobs` folder. The scripts running
-the tests are in the `scripts` folder.
+1. check the following scopes for **Which events would you like to trigger this webhook?**;
 
-## Job image
+   - `Issue comments`
+   - `Pull requests`
+
+We use [metal3-jenkins](https://github.com/metal3-jenkins) GitHub bot account which reports
+job status on a pull request. The bot is a public member of Metal3 GitHub org. To use this bot
+with Jenkins ghprb, a GitHub personal access token was generated with the following scope checked:
+
+![status](images/status.png)
+
+To use the token in Jenkins, `metal3-jenkins-github-token` secret is created in Jenkins credentials.
+See the usage [reference](https://github.com/metal3-io/project-infra/blob/12660dd59d368c86e471007d86ca781cf2539ec9/jenkins/jobs/integration_tests.pipeline#L3).
+
+You can see ghprb logs in [here](https://jenkins.nordix.org/log/GHPRB/), which shows which jobs have used the plugin during their run.
+
+Sometimes, when changing the crendetials related to ghprb, the system might still be using the old credentials. If you see that your changes aren't
+taking affect, you could try to clean up the ghprb cache in Jenkins.
+
+### Some other secrets in Jenkins
+
+- `metal3-jenkins-github-username-token` - stores the username of the metal3-jenkins GitHub bot account.
+   Used by Jenkins Job Builders.
+
+- `metal3-clusterctl-github-token` - stores a GitHub token for use in Metal3
+   integration tests with no permissions. This is to avoid GitHub API limitations for unauthenticated
+   users. Used by [pipeline](https://github.com/metal3-io/project-infra/blob/12660dd59d368c86e471007d86ca781cf2539ec9/jenkins/jobs/integration_tests.pipeline#L88).
+
+Sometimes, your changes might not take effect and you need to flash out Jenkins cache.
```

Comment:
Not sure if we can add it here but [this](https://www.tech693.com/2018/07/jenkins-how-to-clear-cache.html) seems to be good explanation of how to do that.

Reply:
We did it via GUI, not sure neither if we need this.